src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt

```diff
@@ -402,21 +427,83 @@ class AccountControllerTest @Autowired constructor(
         @Test
         fun `should fail to create account with existing email`() {
             println("testAccount: ${objectMapper.writeValueAsString(testAccount)}")
-            mockMvc.post("/accounts/new") {
-                contentType = MediaType.APPLICATION_JSON
-                content = testAccount.toJson()
+
+            val accountPart = MockPart("account", testAccount.toJson().toByteArray())
+            accountPart.headers.contentType = MediaType.APPLICATION_JSON
+
+            mockMvc.multipart("/accounts/new") {
+                part(accountPart)
             }.andExpect { status { isOk() } }
 
-            mockMvc.post("/accounts/new") {
-                contentType = MediaType.APPLICATION_JSON
-                content = testAccount.toJson()
+            mockMvc.multipart("/accounts/new") {
+                part(accountPart)
             }.andExpect {
                 status { isUnprocessableEntity() }
                 content { contentType(MediaType.APPLICATION_JSON) }
                 jsonPath("$.errors.length()") { value(1) }
                 jsonPath("$.errors[0].message") { value("email already exists") }
             }
         }
+
+        @Test
+        fun `should fail to create account with invalid filename extension`() {
+            val accountPart = MockPart("account", testAccount.toJson().toByteArray())
+            accountPart.headers.contentType = MediaType.APPLICATION_JSON
+
+            val file = MockMultipartFile(
+                "photo",
+                "photo.pdf",
+                MediaType.IMAGE_JPEG_VALUE,
+                "content".toByteArray()
+            )
+
+            val uuid: UUID = UUID.randomUUID()
+            val mockedSettings = Mockito.mockStatic(UUID::class.java)
+            Mockito.`when`(UUID.randomUUID()).thenReturn(uuid)
```

Comment:
why do we need mocks here?

Reply:
UUID is a static method that we cannot inject. We could also probably use seeds of random, but I believe this is a more correct way of testing this